File: week2/triangular.py
import sys
def main(): 
	output = 0

	# checks if there are the right number of arguments
        try:
		# converts the first argument to an integer
		in_arg = int(sys.argv[1]);
        except: # (argc != 2)
		# raises an error
		raise Exception("Incorrect arguments.  Usage: triangular [NUM]!")
		
	# iterates over all numbers up to AND INCLUDING the input
        for i in range(in_arg + 1):
		# adds the index to the output
		output = output + i
	
	# prints the output
	print("{}\n".format(output))

if __name__ == "__main__":
        main()

File: week2/triangular.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char **argv) 
{
	// creates and initialises the variables
	int i, input, output;
	i = input = output = 0;

	// checks if there are the right number of arguments
	if (argc == 2)
	{
		// converts the first argument to an integer
		input = atoi(argv[1]);
	}
	else //(argc != 2)
	{
		// raises an error
		fprintf(stderr, "Incorrect arguments.  Usage: triangular [NUM]!");
		// and crashes out
		exit(-1);
	}
	
	// iterates over all numbers up to AND INCLUDING the input
	for (i = 0; i <= input; i++)
	{
		// adds the index to the output
		output = output + i;
	}

	// prints the output
	printf("%d\n", output);
	return 0;
}
